native/packages/react-native-bpk-component-button/src/BpkButton-test.common.js

```diff
@@ -16,96 +16,127 @@
+        />,
       ).toJSON();
-       expect(tree).toMatchSnapshot();
-     });
+      expect(tree).toMatchSnapshot();
+    });
+    it('should support having only an icon', () => {
+      const tree = renderer.create(
+        <BpkButton
+          title="Lorem ipsum"
+          icon={<Image source="../rightarrow_360.png" />}
+          iconOnly
+          onPress={onPressFn}
+        />,
+      ).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
 
-     it('should support overwriting styles', () => {
-       const tree = renderer.create(
-         <BpkButton title="Lorem ipsum" onPress={() => {}} style={{ width: 100 }} />,
+    it('should support overwriting styles', () => {
+      const tree = renderer.create(
+        <BpkButton title="Lorem ipsum" onPress={onPressFn} style={{ width: 100 }} />,
       ).toJSON();
-       expect(tree).toMatchSnapshot();
-     });
+      expect(tree).toMatchSnapshot();
+    });
 
-     BUTTON_TYPES.forEach((buttonType) => {
-       it(`should render correctly with type="${buttonType}"`, () => {
-         const tree = renderer.create(
-           <BpkButton type={buttonType} title="Lorem ipsum" onPress={() => {}} />,
+    BUTTON_TYPES.forEach((buttonType) => {
+      it(`should render correctly with type="${buttonType}"`, () => {
+        const tree = renderer.create(
+          <BpkButton type={buttonType} title="Lorem ipsum" onPress={onPressFn} />,
         ).toJSON();
-         expect(tree).toMatchSnapshot();
-       });
-     });
+        expect(tree).toMatchSnapshot();
+      });
+    });
 
-     it('should accept iconOnly prop when icon prop is supplied', () => {
-       expect(BpkButton.propTypes.icon({
-         iconOnly: true,
-         icon: <Image />,
-       }, 'icon', 'BpkButton')).toBeFalsy();
-     });
+    it('should accept iconOnly prop when icon prop is supplied', () => {
+      expect(BpkButton.propTypes.icon({
+        iconOnly: true,
+        icon: <Image />,
+      }, 'icon', 'BpkButton')).toBeFalsy();
+    });
 
-     it('should accept iconOnly prop when icon prop is supplied', () => {
-       expect(BpkButton.propTypes.icon({
-         iconOnly: true,
-       }, 'icon', 'BpkButton').toString()).toEqual('Error: Invalid prop `icon` supplied to `BpkButton`. When `iconOnly` is enabled, `icon` must be supplied.'); // eslint-disable-line max-len
-     });
-   });
- };
+    it('should accept iconOnly prop when icon prop is supplied', () => {
+      expect(BpkButton.propTypes.icon({
+        iconOnly: true,
+      }, 'icon', 'BpkButton').toString()).toEqual('Error: Invalid prop `icon` supplied to `BpkButton`. When `iconOnly` is enabled, `icon` must be supplied.'); // eslint-disable-line max-len
+    });
+    it('should throw an error for invalid button type', () => {
+      try {
+        renderer.create(
+          <BpkButton
+            title="Lorem ipsum"
+            type="silly"
+            onPress={onPressFn}
+          />,
+        ).toJSON();
+      } catch (exc) {
```

Comment:
This is putting a lot of red output in the console when we run `test`.
Can we use `expect(…).toThrow(…)` instead?

Reply:
unfortunately it does the very same, I think is because it returns the error rather than throw it

Reply:
It does as well. That's annoying
I still think this would be better though:
```
    it('should throw an error for invalid button type', () => {
      expect(() => renderer.create(
        <BpkButton
          title="Lorem ipsum"
          type="silly"
          onPress={onPressFn}
        />,
        )).toThrow('"silly" is not a valid button type. Valid types are primary, featured, secondary, destructive');
    });
```

Reply:
yeah much cleaner, to be honest it was like this in the beginning, but I tried to have the sterr not showing up so I ended up with try/catch